src/js/game/systems/logic_gate.js

```diff
@@ -48,133 +46,84 @@ export class LogicGateSystem extends GameSystemWithFilter {
             assert(slotComp.slots[0].type === enumPinSlotType.logicalEjector, "Slot 0 should be ejector");
 
             slotComp.slots[0].value = result;
-        }
+        });
     }
 
     /**
      * @param {Array<BaseItem|null>} parameters
-     * @returns {BaseItem}
+     * @returns {BooleanItem}
      */
     compute_AND(parameters) {
         assert(parameters.length === 2, "bad parameter count for AND");
 
-        const param1 = parameters[0];
-        const param2 = parameters[1];
-        if (!param1 || !param2) {
-            // Not enough params
-            return BOOL_FALSE_SINGLETON;
-        }
+        const [param1, param2] = parameters;
```

Comment:
There's also a reason I don't use `[a, b] = array` in the whole codebase (with a few exceptions) since it's too slow too. (It gets polyfilled, and the polyfill is very slow)

Reply:
This is what the polyfill looks like:

```
  function Dw(t) {
      var e = t[0],
          r = t[1];
      return Object(hv.d)(e) && Object(hv.d)(r) && e.value && r.value ? hv.b : hv.a
  }
```

It is identical to the code I replaced.